main.go

```diff
@@ -25,6 +25,14 @@ import (
 	"github.com/scylladb/scylla-bench/random"
 )
 
+type (
+	Session interface {
+		Query(string, ...interface{}) *gocql.Query
+	}
+)
+
+type ModeFunc func(session *gocql.Session, testResult *results.TestThreadResult, workload WorkloadGenerator, rateLimiter RateLimiter, validateData bool)
```

Comment:
Can you pleas move out changes not related to purpose of this PR to a separate PR

Reply:
which one, this here is added for tests, to mock gocql.Session

Reply:
exactly, create a PR that enables session mocking and move all these changes there.
Same regarding log.Fatal -> log.Panic